### PR TITLE
SNOW-637927: Check max column length for Binary data type

### PR DESCRIPTION
**Pull request by sfc-gh-tzhang**
We need to check the max column length allowed for the binary data type, otherwise the data will be ingested successfully but not queryable.

### CODE REVIEW

**Comment by sfc-gh-tzhang** [comment]
bump

File: src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java [hunk 1]
```diff
@@ -945,7 +945,12 @@ private float convertRowToArrow(
               break;
             }
           case BINARY:
-            byte[] bytes = DataValidationUtil.validateAndParseBinary(value);
+            String maxLengthString = field.getMetadata().get(COLUMN_BYTE_LENGTH);
```
Comment (sfc-gh-mnaides) [inline src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:948]: @sfc-gh-tzhang Can you confirm that the `COLUMN_BYTE_LENGTH` is what we expect here?

Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:948]: @sfc-gh-mnaides i'm not, based on my testing, byteLength and length always give me the same thing. I assume for char we should use length and for binary for we should use byte length?

**Review by sfc-gh-mnaides — APPROVED** [review]

File: src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java (hunk 1 above)
Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java:948]: verified at the server side that length and byte length are the same value so we should we be good

**Pull request merged**